assets/js/scripts.js

```diff
@@ -1043,6 +1043,30 @@ function formatLootTableLevel(table, rate = 1, level = 0) {
   return result;
 }
 
+const videoModalEl = document.getElementById('video-modal');
+const videoFrameEl = document.getElementById('video-frame');
+videoModalEl.addEventListener('show.bs.modal', function (event) {
+  const button = event.relatedTarget;
+
+  const parentDiv = button.firstAncestorOrSelf((node) => node.tagName === 'DIV');
+  document.getElementById('video-modal-title').textContent =
+    parentDiv.querySelector('h1').innerText || Language.get('map.video');
+
+  const url = button.getAttribute('data-video-url');
+  if (url.includes('youtube')) {
+    const videoId = url.match(/v=([^&]+)/)[1];
+    const startTime = url.match(/t=(\d+)/)?.[1] || 0;
+    videoFrameEl.src = `https://www.youtube.com/embed/${videoId}?autoplay=1&start=${startTime}`;
+  } else if (url.includes('bilibili')) {
+    const videoId = url.match(/\/video\/([^\/?]+)/)[1];
+    const startTime = url.match(/t=(\d+)/)?.[1] || 0;
+    videoFrameEl.src = `https://player.bilibili.com/player.html?bvid=${videoId}&high_quality=1&t=${startTime}`;
+  }

```

Comment:
Just a suggestion. Wouldn't be better to use URL object?

```javascript
  const url = new URL(button.getAttribute('data-video-url')); // Assumming attribute is not null or empty string

  if (['youtube', 'youtu.be'].some(value => url.host.includes(value))) {
    const videoId = url.searchParams.get('v');
    const startTime = url.searchParams.get('t') || 0;
    videoFrameEl.src = `https://www.youtube.com/embed/${videoId}?autoplay=1&start=${startTime}`;
    console.log(videoFrameEl.src);
  } else if (url.host.includes('bilibili')) {
    const [, videoId] = url.pathname.match(/\/video\/([^\/?]+)/); //? /\/video\/(\w+)/
    const startTime = url.searchParams.get('t') || 0;
    videoFrameEl.src = `https://player.bilibili.com/player.html?bvid=${videoId}&high_quality=1&t=${startTime}`;
  }
```

Reply:
Thanks for correction, more modern approach.